be/src/fs/s3/poco_common.cpp

```diff
@@ -0,0 +1,100 @@
+// Copyright 2021-present StarRocks, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "fs/s3/poco_common.h"
+
+#include <Poco/Exception.h>
+#include <fmt/format.h>
+
+#include <memory>
+#include <sstream>
+
+namespace starrocks::poco {
+
+bool isHTTPS(const Poco::URI& uri) {
+    if (uri.getScheme() == "https")
+        return true;
+    else if (uri.getScheme() == "http")
+        return false;
+    else
+        throw std::runtime_error(fmt::format("Unsupported scheme in URI '{}'", uri.toString()));
+}
+
+HTTPSessionPtr makeHTTPSessionImpl(const std::string& host, Poco::UInt16 port, bool https, bool keep_alive,
+                                   bool resolve_host) {
+    HTTPSessionPtr session;
+
+    if (https) {
+        session = std::make_shared<Poco::Net::HTTPSClientSession>(host, port);
+    } else {
+        session = std::make_shared<Poco::Net::HTTPClientSession>(host, port);
+    }
+
+    // doesn't work properly without patch
+    session->setKeepAlive(keep_alive);
+    return session;
+}
+
+void setTimeouts(Poco::Net::HTTPClientSession& session, const ConnectionTimeouts& timeouts) {
+    session.setTimeout(timeouts.connection_timeout, timeouts.send_timeout, timeouts.receive_timeout);
+    session.setKeepAliveTimeout(timeouts.http_keep_alive_timeout);
+}
+
+HTTPSessionPtr makeHTTPSession(const Poco::URI& uri, const ConnectionTimeouts& timeouts, bool resolve_host) {
+    const std::string& host = uri.getHost();
+    Poco::UInt16 port = uri.getPort();
+    bool https = isHTTPS(uri);
+
+    auto session = makeHTTPSessionImpl(host, port, https, false, resolve_host);
+    setTimeouts(*session, timeouts);
+    return session;
+}
+
+std::string getCurrentExceptionMessage() {
+    std::stringstream ss;
```

Comment:
looks like stringstream is not necessary, std::string might be good enough since the fmt::format already convert message to string.

Reply:
yes, will do it.